### PR TITLE
Add a workflow to check if mirrors are offline or outdated

### PR DESCRIPTION
**Pull request by seisman**

### CODE REVIEW

**Review by PaulWessel — APPROVED** [review]
Looks good!

File: .github/workflows/check-mirrors.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#
+# This workflow checks if a mirror is offline or out-of-dated.
+#
+name: Check Mirrors
+on:
+  # uncomment the 'pull_request' line to enable testing in PRs
+  pull_request:
+  schedule:
+    # Cron job att 00:00 on every Sunday
```
Comment (seisman) [inline .github/workflows/check-mirrors.yml:9]:
```suggestion
    # Cron job at 00:00 on every Sunday
```

**Comment by seisman** [comment]
@PaulWessel @Esteban82 In this PR, I added a workflow to check the "Last Modified Date" metadata of the `gmt_data_server.txt` file on all GMT mirrors.

The workflow is scheduled to run at 00:00 on every Sunday, and it will create an issue report if:

- Can't download the file from a mirror => the mirror is likely offline
- The file on the mirror is 14 days older than the file on the oceania server => the mirror didn't sync with the oceania server in the previous 14 days

See https://github.com/GenericMappingTools/gmtserver-admin/issues/217 for an issue report the workflow just created. It's clear that:

- NOAA and Singapore mirrors are offline.
- Brasil and Portugal mirrors are outdated.

My questions:

1. We know the 4 mirrors (maybe NOAA is working) are problematic, but we can do little now. We definitely don't want to see the duplicated issue report every week. So maybe we should check these mirrors every month?

**Comment by PaulWessel** [comment]
Very month seems reasonable.

File: .github/workflows/check-mirrors.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#
+# This workflow checks if a mirror is offline or out-of-dated.
+#
+name: Check Mirrors
+on:
+  # uncomment the 'pull_request' line to enable testing in PRs
+  pull_request:
```
Comment (seisman) [inline .github/workflows/check-mirrors.yml:7]: Disable the workflow in PR:
```suggestion
  # pull_request:
```

**Comment by Esteban82** [comment]
> run at 00:00 on every Sunday

UTC 0? local time in Hawaii??

Is it possible that the workflow is disabled when the repository is not used frequently?

**Comment by seisman** [comment]
> Is it possible that the workflow is disabled when the repository is not used frequently?

Yes, it's possible. But it's very likely that the workflow will always create an issue report every month, which makes the repository "active".

> > run at 00:00 on every Sunday
> 
> UTC 0? local time in Hawaii??

We can change it to any time you like.

**Comment by Esteban82** [comment]
> > UTC 0? local time in Hawaii??
> 
> We can change it to any time you like.

I was just asking. Thanks!

**Pull request merged**